x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtUtil.java

```diff
@@ -0,0 +1,321 @@
+        final boolean ignoreSchemeNameCase
+    ) {
+        final String headerValue = threadContext.getHeader(headerName);
+        if (Strings.hasText(headerValue)) {
+            final String schemeValuePlusSpace = schemeName + " ";
+            if (headerValue.regionMatches(ignoreSchemeNameCase, 0, schemeValuePlusSpace, 0, schemeValuePlusSpace.length())) {
+                final String trimmedSchemeParameters = headerValue.substring(schemeValuePlusSpace.length()).trim();
+                if (Strings.hasText(trimmedSchemeParameters)) {
+                    return new SecureString(trimmedSchemeParameters.toCharArray());
+                }
+            }
+        }
+        return null;
+    }
+
+    // Static method for unit testing. No need to construct a complete RealmConfig with all settings.
+    public static void validateClientAuthorizationSettings(
+        final String clientAuthorizationTypeConfigKey,
+        final String clientAuthorizationType,
+        final String clientAuthorizationSharedSecretConfigKey,
+        final SecureString clientAuthorizationSharedSecret
+    ) throws SettingsException {
+        switch (clientAuthorizationType) {
+            case JwtRealmSettings.CLIENT_AUTHORIZATION_TYPE_SHARED_SECRET:
+                // If type is "SharedSecret", the shared secret value must be set
+                if (Strings.hasText(clientAuthorizationSharedSecret) == false) {
+                    throw new SettingsException(
+                        "Missing setting for ["
+                            + clientAuthorizationSharedSecretConfigKey
+                            + "]. It is required when setting ["
+                            + clientAuthorizationTypeConfigKey
+                            + "] is ["
+                            + JwtRealmSettings.CLIENT_AUTHORIZATION_TYPE_SHARED_SECRET
+                            + "]"
+                    );
+                }
+                break;
+            case JwtRealmSettings.CLIENT_AUTHORIZATION_TYPE_NONE:
+            default:
+                // If type is "None", the shared secret value must not be set
+                if (Strings.hasText(clientAuthorizationSharedSecret)) {
+                    throw new SettingsException(
+                        "Setting ["
+                            + clientAuthorizationSharedSecretConfigKey
+                            + "] is not supported, because setting ["
+                            + clientAuthorizationTypeConfigKey
+                            + "] is ["
+                            + JwtRealmSettings.CLIENT_AUTHORIZATION_TYPE_NONE
+                            + "]"
+                    );
+                }
+                LOGGER.warn(
+                    "Setting ["
+                        + clientAuthorizationSharedSecretConfigKey
+                        + "] value ["
+                        + JwtRealmSettings.CLIENT_AUTHORIZATION_TYPE_NONE
+                        + "] is not recommended for production."
```

Comment:
I think we need a slightly different wording here. I don't have a good suggestion ATM but bringing it up in case. I believe we need to be a bit more specific as to the why, but still keep the message short.  Also, mental note to add a section in the docs, or a `NOTE:` in the settings doc page.

Reply:
A warning was added to docs in the previous PR. I will tweak the wording to mention it is not secure for production.

Reply:
Done